src/python_pybind11/src/MecanumDriveOdometry.cc

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2023 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <string>
+
+#include <gz/math/MecanumDriveOdometry.hh>
+
+#include "MecanumDriveOdometry.hh"
+
+namespace ignition
+{
+namespace math
+{
+namespace python
+{
+void defineMathMecanumDriveOdometry(py::module &m, const std::string &typestr)
+{
+  using Class = gz::math::MecanumDriveOdometry;
+  std::string pyclass_name = typestr;
+  py::class_<Class>(m,
+                    pyclass_name.c_str(),
+                    py::buffer_protocol(),
```

Comment:
Do we need `py::buffer_protocol` here? If so, doesn't it require implementing `def_buffer` (https://pybind11.readthedocs.io/en/stable/advanced/pycpp/numpy.html)

Reply:
I don't think this has been addressed